source/jormungandr/jormungandr/street_network/instance_system.py

```diff
@@ -0,0 +1,151 @@
+# Copyright (c) 2001-2016, Canal TP and/or its affiliates. All rights reserved.
+#
+# This file is part of Navitia,
+#     the software to build cool stuff with public transport.
+#
+# Hope you'll enjoy and contribute to this project,
+#     powered by Canal TP (www.canaltp.fr).
+# Help us simplify mobility and open public transport:
+#     a non ending quest to the responsive locomotion way of traveling!
+#
+# LICENCE: This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Stay tuned using
+# twitter @navitia
+# IRC #navitia on freenode
+# https://groups.google.com/d/forum/navitia
+# www.navitia.io
+
+import pybreaker
+import requests as requests
+from dateutil import parser
+import time
+import datetime
+import logging
+from jormungandr import utils
+from ride_sharing_journey import RideSharingJourney
+
+
+def _record_external_failure(message, sn_system_id):
+    utils.record_external_failure(message, 'streetnetwork', sn_system_id)
+
+
+def _call_service(url, api_key, sn_system_id):
+    logger = logging.getLogger(__name__)
+    breaker = pybreaker.CircuitBreaker(fail_max=100, reset_timeout=1000)
+    headers = {u'Authorization': u'apiKey {}'.format(api_key)}
+    try:
+        return breaker.call(requests.get, url=url, headers=headers, timeout=1000)
+    except pybreaker.CircuitBreakerError as e:
+        logger.error('Ride sharing service dead (error: {})'.format(e))
+        _record_external_failure('circuit breaker open', sn_system_id)
```

Comment:
It's better to use exception to handle that like it has been done in the realtime schedules part. 
But we will do the refactoring latter.